@AH-64D Apache Longbow/Addons/fza_ah64_ihadss/functions/fn_linearMotionCompensator.sqf

```diff
@@ -0,0 +1,133 @@
+/* ----------------------------------------------------------------------------
+Function: fza_ihadss_fnc_linearMotionCompensator
+
+Description:
+    Per Frame handler of the LMC
+
+Parameters:
+    _heli - The helicopter to act on
+
+Returns:
+
+Examples:
+
+Author:
+    Snow(Dryden), Ampersand
+---------------------------------------------------------------------------- */
+#define INPUT_SENS 0.05 // Tweak this for responsiveness? - Ampersand
+#define INPUT_MAX 5.0
+
+params ["_heli"];
+
+if !(_heli getVariable "fza_ah64_LmcActive") exitwith {
+    _heli setVariable ["fza_ah64_lmcConstant", [0, 0]];
+    _heli setVariable ["fza_ah64_lmcStartRange", -1];
+    _heli setVariable ["fza_ah64_lmcRange", 1000];
+    _heli setVariable ["fza_ah64_lmcPosition", []];
+};
+private _sight = [_heli, "fza_ah64_sight"] call fza_fnc_getSeatVariable;
+if (_sight != SIGHT_TADS || !(local gunner _heli)) exitwith {};
+
+(_heli getVariable "fza_ah64_lmcConstant") params ["_azimuthC", "_elevationC"];
+private _lmcStartRange = _heli getVariable "fza_ah64_lmcStartRange";
+private _lmcPosition = _heli getVariable "fza_ah64_lmcPosition";
+
+private _fovVal = fza_ah64_tadsFOVs select (_heli getTurretOpticsMode [0]);
+private _inputX = [((((inputAction "AimRight" - inputAction "AimLeft")) * INPUT_SENS) * _fovVal + _azimuthC), -INPUT_MAX, INPUT_MAX] call BIS_fnc_clamp;
+private _inputY = [(((inputAction "AimUp" - inputAction "AimDown") * INPUT_SENS) * _fovVal + _elevationC), -INPUT_MAX, INPUT_MAX] call BIS_fnc_clamp;
+
+_heli setVariable ["fza_ah64_lmcConstant", [_inputX, _inputY]];
+
+#ifdef __A3_DEBUG__
+drawIcon3D [
+    "a3\ui_f\data\Map\Markers\Military\circle_CA.paa",
+    [1, 1, 1, 1],
+    positionCameraToWorld [_inputX, _inputY, 10],
+    1, 1, 0
+];
+#endif
+
+//AUTO RANGING USING ATL ON 2D Model
+private _TadsPosition  = _heli modelToWorldVisualWorld (_heli selectionPosition "laserEnd");
+private _TadsAimPos    = _heli modelToWorldVisualWorld (_heli selectionPosition "laserBegin");
+private _tadsDirection = (_TadsPosition vectorFromTo _TadsAimPos) vectorMultiply 50000;
+_tadsDirection call CBA_fnc_vect2Polar Params ["","","_elevation"];
+private _autorange = [(_TadsPosition)#2 /sin(-_elevation),0,50000] call BIS_fnc_clamp;
+
+private _range = -1;
```

Comment:
The range source logic here looks good, but I would prefer for us to have a separate function that does this for reuse

Reply:
Will be left for a seperate PR